### PR TITLE
ISD AS Identity

### PR DESCRIPTION
**Pull request by juagargi**
Expanded the section.

### CODE REVIEW

File: draft-meynell-panrg-scion-research-questions.md [hunk 1]
```diff
@@ -196,9 +196,34 @@ This draft assumes the reader is familiar with some of the fundamental concepts
...
+  Note that not all combinations of ASes might be possible as ISDs:
+  An ISD MUST be connected via at least one core link to the rest of the ISD connected graph.
+  Otherwise the beaconing process does not transfer trust information among
+  the connected graph of existing ISDs and this ISD in question.
+  This reduces the absolute number of possible ISDs to those that satisfy a
+  strict direct connectivity requirement (hence they could become core ASes in
+  the new ISD). It is still super-exponential asymptotically.
+  * Per geographic area?
+  * Per legal jurisdiction?
+  * Per capacity tier?
```
Comment (tzaeschke) [inline draft-meynell-panrg-scion-research-questions.md:224]: Can you briefly motivate what the benefit would be for someone to create an ISD to indicate capacity?
Also, since an AS has a given capacity, I guess it should only ever be in one of those ISDs, never in multiple ones?

File: draft-meynell-panrg-scion-research-questions.md [hunk 2]
```diff
@@ -196,9 +196,34 @@ This draft assumes the reader is familiar with some of the fundamental concepts
 
 ## ISD, AS identity
 
+The SCION protocol specifies 16 bits and 48 bits to identify the ISD and AS.
+These identification is, at this moment, necessary at the data-plane level,
```
[the remark this answers is not in the document]
Comment (juagargi) [reply draft-meynell-panrg-scion-research-questions.md:200]: The question posed in this section relates to how many ISDs and ASes we expect, and what upper bound we have for them. E.g. 16 bits for ISD numbers allow only for ~16K distinct ISDs.

File: draft-meynell-panrg-scion-research-questions.md [hunk 3]
```diff
@@ -196,9 +196,34 @@ This draft assumes the reader is familiar with some of the fundamental concepts
...
+The following questions arise: (not comprehensive)
+
 * How many ASes do we expect in the SCION network model?
-* How many ISDs? What is the ontology of an ISD? Per geographic area only?
-* One AS belongs to many ISDs?
+* Can one AS belong to many ISDs?
+* Are AS numbers unique themselves? Or only unique in combination with an ISD?
+* How many ISDs do we expect?
+* What is the ontology of an ISD?
+  Note that not all combinations of ASes might be possible as ISDs:
```
[the remark this answers is not in the document]
Comment (juagargi) [reply draft-meynell-panrg-scion-research-questions.md:215]: Yes, it refers to an AS being part of multiple ISDs.

File: draft-meynell-panrg-scion-research-questions.md [hunk 4]
```diff
@@ -196,9 +196,34 @@ This draft assumes the reader is familiar with some of the fundamental concepts
...
+  An ISD MUST be connected via at least one core link to the rest of the ISD connected graph.
+  Otherwise the beaconing process does not transfer trust information among
+  the connected graph of existing ISDs and this ISD in question.
+  This reduces the absolute number of possible ISDs to those that satisfy a
+  strict direct connectivity requirement (hence they could become core ASes in
+  the new ISD). It is still super-exponential asymptotically.
```
[the remark this answers is not in the document]
Comment (tzaeschke) [reply draft-meynell-panrg-scion-research-questions.md:221]: Well, there are probably infinitely many thinkable "types" :-). Maybe the point is that "super-exponential asymptotically" sound very specific and mathematically precise.

I think the point of the section is that should make sure (e.g. provide an adverse incentive or maybe a strict specification) that prevents ASes to join to many ISD, or alternatively, to provide an adverse incentive (or a strict specification) that prevents creation of ISDs?

Maybe we can emphasize somewhere that ISD should really only be created to signify "trust" and no other "types" of commonalities? That would mitigate the problem a little bit.

File: draft-meynell-panrg-scion-research-questions.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (tzaeschke) [reply draft-meynell-panrg-scion-research-questions.md]: Maybe it can be rephrased a bit? I think I misunderstood the meaning of "all". I think what it means is that all (=arbitrary) combinations should definitely be possible, but not (all) at the same time. 
Does that make sense?

Actually, I also think the word "combinations" is a bit misleading. I think it doesn't really matter how many combinations are available. If ISD start being used as "types", many of them may contain an identical set of ASes. We may still run out of ISD numbers, even if the number of actually used combinations is quite low....

File: draft-meynell-panrg-scion-research-questions.md (hunk 1 above)
Comment (tzaeschke) [reply draft-meynell-panrg-scion-research-questions.md:224]: > but there might be many e.g. "Tier-2" ISDs, since not all ASes that fall into the category tier-2 can directly connect with each other.

Is that the case? I didn't think a direct connection is a requirement? For example in SCIERA, I guess not all Universities or core ASes have a direct connection, even if they are all in the same ISD?

That question aside, I am still not sure what the motivation would be to create ISDs for capacity tiers. 
However, I won't object if you leave it as is :-)

File: draft-meynell-panrg-scion-research-questions.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (tzaeschke) [reply draft-meynell-panrg-scion-research-questions.md]: I think from the text it sound like "at the moment" refers to being "necessary on the data-plane level".

Maybe rephrase this section to:
``
[At the moment] the SCION protocol specifies 16 bits and 48 bits to identify the ISD and AS.
This identification is used, at the data-plane level, in every data packet to fully address the sender and receiver, and at the control-plane level, to identify the PCB sender and hops.`
```

**Review by tzaeschke — APPROVED** [review]
LGTM % open comments

File: draft-meynell-panrg-scion-research-questions.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (juagargi) [reply draft-meynell-panrg-scion-research-questions.md:221]: Correct, using trust as the main driver to create ISDs is a conclusion I would like to reach through the discussion.
It is also true that there might be an arbitrary big number of "types". And maybe the discussion goes to that point too, describing what "types" are acceptable to create ISDs.
Still, due to the control plane protocol, core ASes have to beacon directly to their neighbors. Whether that _directly_ means an L2 or something else, that is also open to debate.

File: draft-meynell-panrg-scion-research-questions.md [hunk 5]
```diff
@@ -196,9 +196,34 @@ This draft assumes the reader is familiar with some of the fundamental concepts
 
 ## ISD, AS identity
 
+The SCION protocol specifies 16 bits and 48 bits to identify the ISD and AS.
```
Comment (knmeynell) [inline draft-meynell-panrg-scion-research-questions.md:199]:
```suggestion
The SCION protocol specifies 16 bits and 48 bits to identify the ISD and AS respectively.
```

Comment (knmeynell) [reply draft-meynell-panrg-scion-research-questions.md:199]: What is the correct terminology here for the 16+48 bit identifier?

File: draft-meynell-panrg-scion-research-questions.md [hunk 6]
```diff
@@ -196,9 +196,34 @@ This draft assumes the reader is familiar with some of the fundamental concepts
...
+This identification is, at this moment, necessary at the data-plane level.
+It is written in each one of the data packets to fully address the sender and receiver,
+as well as at the control-plane level, to identify the PCB sender and hops.
+
+While 48 bits seem sufficient to accommodate a large number of ASes,
```
Comment (knmeynell) [inline draft-meynell-panrg-scion-research-questions.md:204]:
```suggestion
Whilst 48 bits for the AS will accommodate up to 2.81475e14 assignments which is likely to be more than sufficient for the future,
```

File: draft-meynell-panrg-scion-research-questions.md [hunk 7]
```diff
@@ -196,9 +196,34 @@ This draft assumes the reader is familiar with some of the fundamental concepts
...
+using 16 bits for the ISD might not be.
```
Comment (knmeynell) [inline draft-meynell-panrg-scion-research-questions.md:205]: using 16 bits for the ISD only offers 65,536 possible assignments which is conceivably insufficient.

File: draft-meynell-panrg-scion-research-questions.md [hunk 8]
```diff
@@ -196,9 +196,34 @@ This draft assumes the reader is familiar with some of the fundamental concepts
...
+Additionally, the justification of these figures is lacking at the moment.
```
Comment (knmeynell) [inline draft-meynell-panrg-scion-research-questions.md:206]: Can this be better clarified? Is there any more information about why these values were chosen.

File: draft-meynell-panrg-scion-research-questions.md [hunk 9]
```diff
@@ -196,9 +196,34 @@ This draft assumes the reader is familiar with some of the fundamental concepts
...
+
 * How many ASes do we expect in the SCION network model?
-* How many ISDs? What is the ontology of an ISD? Per geographic area only?
-* One AS belongs to many ISDs?
+* Can one AS belong to many ISDs?
+* Are AS numbers unique themselves? Or only unique in combination with an ISD?
+* How many ISDs do we expect?
+* What is the ontology of an ISD?
+  Note that the entire set of all combinations of ASes might be the complete set of ISDs:
+  An ISD MUST be connected via at least one core link to the rest of the ISD connected graph.
```
Comment (knmeynell) [inline draft-meynell-panrg-scion-research-questions.md:216]: What is the meaning of 'graph' here?

File: draft-meynell-panrg-scion-research-questions.md [hunk 10]
```diff
@@ -196,9 +196,34 @@ This draft assumes the reader is familiar with some of the fundamental concepts
...
+  Otherwise the beaconing process does not transfer trust information among
+  the connected graph of existing ISDs and this ISD in question.
+  This reduces the absolute number of possible ISDs to those that satisfy a
+  strict direct connectivity requirement (hence they could become core ASes in
+  the new ISD). It is still super-exponential asymptotically.
```
Comment (knmeynell) [inline draft-meynell-panrg-scion-research-questions.md:221]: This terminology needs some work.

**Review by knmeynell — CHANGES_REQUESTED** [review]

File: draft-meynell-panrg-scion-research-questions.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (tzaeschke) [reply draft-meynell-panrg-scion-research-questions.md:221]: > Still, due to the control plane protocol, core ASes have to beacon directly to their neighbors. Whether that directly means an L2 or something else, that is also open to debate.

Yes, they have to _beacon_  to neighbors, but I think an core-AS has no obligation to have a direct connection to other core-AS in the same ISD. They can still beacon to neighbors in other ISDs. Or not?

[the remark this answers is not in the document]
Comment (nicorusti) [reply draft-meynell-panrg-scion-research-questions.md:221]: I think we will be talking a bit more about ISD creation and governance in the deployment draft, so perhaps in this draft we can only focus on wether 65k ISDs is enough.

[the remark this answers is not in the document]
Comment (juagargi) [reply draft-meynell-panrg-scion-research-questions.md:221]: I reworked the paragraphs.

File: draft-meynell-panrg-scion-research-questions.md (hunk 8 above)
Comment (juagargi) [reply draft-meynell-panrg-scion-research-questions.md:206]: Honestly I don't know why they were chosen as 16-48. In the book we also don't say.

Comment (nicorusti) [reply draft-meynell-panrg-scion-research-questions.md:206]: >  which is conceivably insufficient 

Wether this is sufficient or not depends on how these would be assigned.. as you say we don't have concrete justifications on these figures, so we also have no justifications wether this is sufficient or not. I'd therefore change language a bit from: 
> using 16 bits for the ISD only offers 65,536 possible assignments which is conceivably insufficient.
Additionally, the justification of these figures is lacking at the moment.

To
> using 16 bits for the ISD only offers 65,536 possible assignments. Further investigation on whether thisis sufficient is needed.

File: draft-meynell-panrg-scion-research-questions.md [hunk 11]
```diff
@@ -196,9 +196,31 @@ This draft assumes the reader is familiar with some of the fundamental concepts
...
+* Can one AS belong to many ISDs?
+* Are AS numbers unique themselves? Or only unique in combination with an ISD?
+* How many ISDs do we expect?
+* What is the ontology of an ISD?
+  * Per geographic area?
+  * Per legal jurisdiction?
+  * Per capacity tier?
+  * Per "type"? (meaning: any grouping that makes sense to their members)
+  * Possible combinations of any previous "types"?
+* Note that ISDs require CORE links to other ISDs.
```
Comment (nicorusti) [inline draft-meynell-panrg-scion-research-questions.md:220]: We don't really mandate connectivity in the specification. Is there a reason why you say "require"?

Comment (juagargi) [reply draft-meynell-panrg-scion-research-questions.md:220]: Core AS from ISD1 MUST send PCBs to core AS from ISD2.
That's where the connectivity requirement comes from.
But I don't know if I should change the wording, so that it is clear that both ISDs must be able to communicate via their core ASes, in principle without intermediaries.

File: draft-meynell-panrg-scion-research-questions.md (hunk 5 above)
Comment (nicorusti) [reply draft-meynell-panrg-scion-research-questions.md]: We use `ISD-AS` in the drafts, let's keep it consistent

**Review by nicorusti — APPROVED** [review]
